agent/statechange/statechange.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2014-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package statechange
+
+const (
+	// ContainerEvent is used to define the container state transition events
+	// emitted by the engine
+	ContainerEvent Event = iota
```

Comment:
we decided to go with enums to type the events in this case instead of using reflect package.